patterns/behavioral/strategy.py

```diff
@@ -7,42 +7,77 @@
 Enables selecting an algorithm at runtime.
 """
 
+from __future__ import annotations
+
+from typing import Callable, Type
+
+
+class DiscountStrategyValidator:  # Descriptor class for check perform
+    def validate(self, obj: Order, value: Callable) -> bool:
+        try:
+            if obj.price - value(obj) < 0:
+                raise ValueError(f"Discount cannot be applied due to negative price resulting. {value.__name__}")
+            return True
+        except ValueError as ex:
+            print(str(ex))
+            return False
+
+    def __set_name__(self, owner, name: str) -> None:
+        self.private_name = '_' + name
+
+    def __set__(self, obj: Order, value: Callable = None) -> None:
```

Comment:
Please run against a linter, AFAIR whitespaces in default arguments are not recommended

Reply:
Got E252 missing whitespace around parameter equals
Trying to resolve it

Reply:
Turns out it is acceptable to leave whitespaces in this case